src/devices/Sn74hc595/Sn74hc595.cs

```diff
@@ -0,0 +1,354 @@
+        {
+            _spiDevice = spiDevice;
+        }
+
+        /// <summary>
+        /// Count of shift units.
+        /// The number of (daisy-chained) shift registers. Minimum is 1.
+        /// Not the count of registers on a single unit.
+        /// </summary>
+        public int DeviceCount => _deviceCount;
+
+        /// <summary>
+        /// Count of total bits / registers across all (daisy-chained) shift registers.
+        /// Minimum is 8.
+        /// </summary>
+        public int Bits => _deviceCount * 8;
+
+        /// <summary>
+        /// Reports if Sn74hc595 is controlled with SPI.
+        /// </summary>
+        public bool UsesSpi => _spiDevice is object;
+
+        /// <summary>
+        /// Reports if Sn74hc595 is controlled with GPIO.
+        /// </summary>
+        public bool UsesGpio => _controller is object;
+
+        /// <summary>
+        /// Clear storage registers.
+        /// Requires use of GPIO controller.
+        /// </summary>
+        public void ClearStorage()
+        {
+            if (_controller is null || _pinMapping.SrClr == 0)
+            {
+                throw new ArgumentNullException($"{nameof(ClearStorage)}: GpioController was not provided or {nameof(_pinMapping.SrClr)} not mapped to pin");
+            }
+
+            _controller.Write(_pinMapping.SrClr, 0);
+            _controller.Write(_pinMapping.SrClr, 1);
+        }
+
+        /// <summary>
+        /// Shifts zeros.
+        /// Will dim all connected LEDs, for example.
+        /// Supports GPIO controller or SPI device.
+        /// </summary>
+        public void ShiftClear()
+        {
+            for (int i = 0; i < DeviceCount; i++)
+            {
+                ShiftByte(0);
+            }
+        }
+
+        /// <summary>
+        /// Shifts single value to next register
+        /// Does not perform latch.
+        /// Requires use of GPIO controller.
+        /// </summary>
+        public void ShiftBit(PinValue value)
+        {
+            if (_controller is null || _pinMapping.Data == 0)
+            {
+                throw new ArgumentNullException($"{nameof(ShiftBit)}: GpioController was not provided or {nameof(_pinMapping.Data)} not mapped to pin");
+            }
+
+            _controller.Write(_data, value);
+            _controller.Write(_srclk, 1);
+            _controller.Write(_data, 0);
+            _controller.Write(_srclk, 0);
+        }
+
+        /// <summary>
+        /// Shifts a byte -- 8 bits -- to the 8 registers.
+        /// Pushes / overwrites any existing values.
+        /// Latches by default.
+        /// </summary>
+        public void ShiftByte(byte value, bool latch = true)
+        {
+            if (_spiDevice is object)
+            {
+                _spiDevice.WriteByte(value);
+                return;
+            }
+
+            for (int i = 0; i < 8; i++)
+            {
+                int data = (128 >> i) & value;
```

Comment:
Nit: instead of using 128, can you use `0b_1000_0000` in order to make it clear what is happening here?

Reply:
Fixed